acceptance/tests/puppet_device_test.rb

```diff
@@ -1,74 +1,81 @@
 test_name "puppet device is able to run and configure a node"
 
+tag 'server'
+
 teardown do
   on(master, "[ -f /etc/puppetlabs/puppet/puppet.conf.bak ] && mv /etc/puppetlabs/puppet/puppet.conf.bak /etc/puppetlabs/puppet/puppet.conf")
   # revert permission changes to reset state for other tests
   on(master, "puppet agent -t --server #{master}")
 end
 
-# set sensible defaults
-on(master, "cp /etc/puppetlabs/puppet/puppet.conf /etc/puppetlabs/puppet/puppet.conf.bak")
-on(master, "puppet config set server $(hostname --fqdn)")
-
-codedir = "#{master.puppet['codedir']}/modules"
-confdir = master.puppet['confdir']
-common_options="--config /tmp/puppet.conf --debug --trace -w0"
-
-on(master, puppet("module install --target-dir #{codedir} puppetlabs-test_device"))
-on(master, puppet("module install --target-dir #{codedir} puppetlabs-device_manager"))
-on(master, puppet("module install --target-dir #{codedir} puppetlabs-resource_api"))
-
-apply_manifest_on(master, <<MANIFEST, :catch_failures => true)
-# create and configure test user to trigger PUP-9642
-user {
-  'test':
-    ensure => 'present',
-}
-
-# testfile for --apply
-file {
-  '/tmp/spinner.pp':
-    content => "spinner { '999': ensure => 'absent' }"
-}
-
-# install and configure the Resource API
-service { 'puppetserver': }
-include 'resource_api::server', 'resource_api::agent'
+with_puppet_running_on(master, {}) do
+  # set sensible defaults
+  on(master, "cp /etc/puppetlabs/puppet/puppet.conf /etc/puppetlabs/puppet/puppet.conf.bak")
+  on(master, "puppet config set server $(hostname --fqdn)")
+
+  codedir = "#{master.puppet['codedir']}/modules"
+  confdir = master.puppet['confdir']
+  common_options="--config /tmp/puppet.conf --debug --trace -w0"
+
+  on(master, puppet("module install --target-dir #{codedir} puppetlabs-test_device"))
+  on(master, puppet("module install --target-dir #{codedir} puppetlabs-device_manager"))
+  on(master, puppet("module install --target-dir #{codedir} puppetlabs-resource_api"))
+
+  apply_manifest_on(master, <<MANIFEST, :catch_failures => true)
+  # create and configure test user to trigger PUP-9642
+  user {
+    'test':
+      ensure => 'present',
+  }
+
+  # testfile for --apply
+  file {
+    '/tmp/spinner.pp':
+      content => "spinner { '999': ensure => 'absent' }"
+  }
+
+  # install and configure the Resource API
+  service { 'puppetserver': }
```

Comment:
I think this is unnecessary now

Reply:
Yeah I guess in 6.x it should be.

Reply:
I think it ensures that puppetserver is running as a service, which won't work in PE and is unneeded with the `with_puppet_running_on` now.